core/io/resource_loader.cpp

```diff
@@ -1196,6 +1196,17 @@ bool ResourceLoader::has_custom_uid_support(const String &p_path) {
 	return false;
 }
 
+bool ResourceLoader::should_create_uid_file(const String &p_path) {
+	String local_path = _validate_local_path(p_path);
+
+	for (int i = 0; i < loader_count; i++) {
+		if (loader[i]->recognize_path(local_path)) {
+			return !loader[i]->has_custom_uid_support() && !FileAccess::exists(p_path + ".uid");
```

Comment:
`!FileAccess::exists(p_path + ".uid")` doesn't seem to be affected by the loop, so maybe pull it outside ?

Reply:
Checking whether file exists only makes sense if it can be created. Otherwise it's irrelevant.

Reply:
If `recognize_path` is lightweight then it's probably better like that. But if there is a lot of loaders, or `recognize_path` is expensive, then it may make sense to early out if `FileAccess::exists(p_path + ".uid")` is true.

Reply:
Ok makes sense.